ipaserver/plugins/config.py

```diff
@@ -85,6 +85,13 @@
 
 register = Registry()
 
+
+def validate_search_records_limit(ugettext, value):
+    """Check if value in greater than a realistic minimum."""
+    if 0 < value < 10:
```

Comment:
This also allows < -1 which is not allowed. Only values >= -1 should be allowed.

I'd argue that 10 seconds is far too high. I might jump to 4 seconds but remember that for any one command there can be multiple searches performed so the total search limit is the accumulation of all searches. We tend to have an aggressive timeout so users won't break out of commands thinking something is wrong.

Reply:
> This also allows < -1 which is not allowed. Only values >= -1 should be allowed.

Your are right, but those values will be rejected by the `minvalue=-1` argument. Checking for values >= -1 here will make it more explicitly for sure, but will make `minvalue` unnecessary, right? So it could be removed and this function remains as the only validation method, am I right?

> I'd argue that 10 seconds is far too high. I might jump to 4 seconds but remember that for any one command there can be multiple searches performed so the total search limit is the accumulation of all searches. We tend to have an aggressive timeout so users won't break out of commands thinking something is wrong.

I don't know if right, but I think this config sets the number of records returned by search (`ipasearchrecordslimit`), not the time which is defined by the `ipasearchtimelimit` command.